app/controllers/registrations_controller.rb

```diff
@@ -30,7 +30,11 @@ def create
       yield resource if block_given?
       if resource.persisted?
         update_first_user_permissions(resource)
-        redirect_to "/confirm-email?email=#{CGI.escape(resource.email)}"
+        if SiteConfig.smtp_enabled?
+          redirect_to "/confirm-email?email=#{CGI.escape(resource.email)}"
+        else
+          redirect_to root_path
```

Comment:
I'll re-read your PR description - if the user is still not confirmed the redirect to root path has them logged in correctly and just skips the verification step? It's surprising this does what we need - but feels equivalent to the experiment I followed manually when trying to set up a user on self-host (just navigate to / and ignore the email verification page).